src/attributes.jl

```diff
@@ -180,6 +180,21 @@ function h5readattr(filename, name::AbstractString)
     dat
 end
 
+"""
+    num_attrs()
+
+Retrieve the number of attributes from an object.
+
+See [`API.h5o_get_info`](@ref).
+"""
+function num_attrs(obj)
+    info = @static if API._libhdf5_build_ver < v"1.12.0"
+        API.h5o_get_info(checkvalid(obj))
+    else
+        API.h5o_get_info(checkvalid(obj), API.H5O_INFO_NUM_ATTRS)
+    end
```

Comment:
For pre v1.12.0 HDF5, this uses `H5Oget_info1`. For post v1.12.0, this uses `H5Oget_info3`.